### PR TITLE
eth/tracers: fix state hooks in API

### PR DESCRIPTION
**Pull request by s1na**
When a tx/block was being traced through the API the state hooks weren't being called as they should. This is due to #30745 moving the hooked statedb one level up in the state processor. This PR fixes that.

### CODE REVIEW

**Comment by s1na** [comment]
@rjl493456442 @holiman PTAL

File: eth/tracers/api_test.go [hunk 1]
```diff
@@ -187,6 +188,90 @@ func (b *testBackend) StateAtTransaction(ctx context.Context, block *types.Block
...
 }
 
+type stateTracer struct {
+	Balance map[common.Address]*hexutil.Big
+	Nonce   map[common.Address]hexutil.Uint64
+	Storage map[common.Address]map[common.Hash]common.Hash
+}
+
+func newStateTracer(ctx *Context, cfg json.RawMessage, chainCfg *params.ChainConfig) (*Tracer, error) {
+	t := &stateTracer{Balance: make(map[common.Address]*hexutil.Big), Nonce: make(map[common.Address]hexutil.Uint64), Storage: make(map[common.Address]map[common.Hash]common.Hash)}
```
Comment (rjl493456442) [inline eth/tracers/api_test.go:198]: nitpick, can you change the initialization style a bit to 

```go
    t := &stateTracer{
        Balance: make(map[common.Address]*hexutil.Big), 
        Nonce:   make(map[common.Address]hexutil.Uint64), 
        Storage: make(map[common.Address]map[common.Hash]common.Hash),
    }
```

it's easier to read

**Review by rjl493456442 — APPROVED** [review]
lgtm, sorry for breaking the tracer hooks.

File: eth/tracers/api_test.go (hunk 1 above)
Comment (s1na) [reply eth/tracers/api_test.go]: Yep agreed 👍

File: eth/tracers/api_test.go [hunk 2]
```diff
@@ -187,6 +188,94 @@ func (b *testBackend) StateAtTransaction(ctx context.Context, block *types.Block
...
+	Balance map[common.Address]*hexutil.Big
+	Nonce   map[common.Address]hexutil.Uint64
+	Storage map[common.Address]map[common.Hash]common.Hash
+}
+
+func newStateTracer(ctx *Context, cfg json.RawMessage, chainCfg *params.ChainConfig) (*Tracer, error) {
+	t := &stateTracer{
+		Balance: make(map[common.Address]*hexutil.Big),
+		Nonce: make(map[common.Address]hexutil.Uint64),
+		Storage: make(map[common.Address]map[common.Hash]common.Hash)
```
Comment (holiman) [inline eth/tracers/api_test.go:201]:
```suggestion
		Storage: make(map[common.Address]map[common.Hash]common.Hash),
```

**Pull request merged**